### PR TITLE
Implement S.add and S.sub

### PR DESCRIPTION
**Pull request by svozza**
As discussed in #127 .

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -2195,6 +2196,36 @@
...
+  //# add :: FiniteNumber -> FiniteNumber -> FiniteNumber
+  //.
+  //. Adds two finite numbers together.
+  //.
+  //. ```javascript
+  //. > S.add(1, 1);
+  //. 2
+  //. ```
+  S.add =
+    def('add',
```
Comment (davidchambers) [inline index.js]: :arrow_left: :arrow_left:

**Comment by davidchambers** [comment]
Is there value in using the same names as [sanctuary-int][1], do you think? We use `add` and `sub` in that project.


[1]: https://github.com/plaid/sanctuary-int

**Comment by svozza** [comment]
Yeah, I guess it doesn't do any harm to be consistent. I was just following the naming convention from Ramda.

File: index.js [hunk 2]
```diff
@@ -2195,6 +2196,36 @@
...
+  //. > S.add(1, 1);
+  //. 2
+  //. ```
+  S.add =
+  def('add',
+      {},
+      [$.FiniteNumber, $.FiniteNumber, $.FiniteNumber],
+      function(a, b) { return a + b; });
+
+  //# subtract :: FiniteNumber -> FiniteNumber -> FiniteNumber
```
Comment (davidchambers) [inline index.js]: `s/subtract/sub/`

Could you update the commit message and pull request title to match? :)

Comment (svozza) [inline index.js]: Way ahead of you. :wink:

File: test/index.js [hunk 3]
```diff
@@ -3086,6 +3106,76 @@ describe('object', function() {
...
+      assert.throws(function() { S.sub('xxx', 1); },
+                    errorEq(TypeError,
+                            '‘sub’ expected a value of type FiniteNumber ' +
+                            'as its first argument; received "xxx"'));
+
+      assert.throws(function() { S.sub(1, 'xxx'); },
+                    errorEq(TypeError,
+                            '‘sub’ expected a value of type FiniteNumber ' +
+                            'as its second argument; received "xxx"'));
+    });
```
Comment (davidchambers) [inline test/index.js]: It would be nice to use an infinite number in one of these tests. ∞

File: index.js [hunk 4]
```diff
@@ -2195,6 +2196,36 @@
         return filter(is(type), Just(x));
       });
 
+  //. ### Numeric
+
+  //# add :: FiniteNumber -> FiniteNumber -> FiniteNumber
+  //.
+  //. Adds two finite numbers together.
```
Comment (davidchambers) [inline index.js]: For the descriptions, I like:

  - _Returns the sum of two (finite) numbers._
  - _Returns the difference between two (finite) numbers._

File: index.js [hunk 5]
```diff
@@ -2195,6 +2196,36 @@
         return filter(is(type), Just(x));
       });
 
+  //. ### Numeric
```
Comment (davidchambers) [inline index.js]: I'd prefer __Number__, since the other titles—except __Classify__ which we could rename __Classification__—are nouns.

**Comment by svozza** [comment]
:zap:  x 3

File: index.js [hunk 6]
```diff
@@ -2195,6 +2196,36 @@
...
       });
 
+  //. ### Number
+
+  //# add :: FiniteNumber -> FiniteNumber -> FiniteNumber
+  //.
+  //. Returns the sum of two (finite) numbers.
+  //.
+  //. ```javascript
+  //. > S.add(1, 1);
```
Comment (davidchambers) [inline index.js]: `s/;//`

File: test/index.js [hunk 7]
```diff
@@ -3086,6 +3106,96 @@ describe('object', function() {
 
 });
 
+describe('numeric', function() {
```
Comment (davidchambers) [inline test/index.js]: `s/numeric/number/`

**Comment by svozza** [comment]
:zap:  x 2

**Comment by davidchambers** [comment]
:deciduous_tree:

**Pull request merged**